tests/unit/feature_flags/test_feature_flags.py

```diff
@@ -0,0 +1,1406 @@
+    rule = {schema.CONDITIONS_KEY: []}
+    rules_context = {}
+    feature_flags = init_feature_flags(mocker, {})
+
+    # WHEN calling _evaluate_conditions
+    result = feature_flags._evaluate_conditions(
+        rule_name="dummy",
+        feature_name="dummy",
+        rule=rule,
+        context=rules_context,
+    )
+
+    # THEN return False
+    assert result is False
+
+
+def test_features_jmespath_envelope(mocker):
+    expected_value = True
+    mocked_app_config_schema = {"features": {"my_feature": {"default": expected_value}}}
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, envelope="features")
+    toggle = feature_flags.evaluate(name="my_feature", context={}, default=False)
+    assert toggle == expected_value
+
+
+# test_match_rule_with_equals_action
+def test_match_condition_with_dict_value(mocker):
+    expected_value = True
+    mocked_app_config_schema = {
+        "my_feature": {
+            "default": False,
+            "rules": {
+                "tenant id is 6 and username is lessa": {
+                    "when_match": expected_value,
+                    "conditions": [
+                        {
+                            "action": RuleAction.EQUALS.value,
+                            "key": "tenant",
+                            "value": {"tenant_id": "6", "username": "lessa"},
+                        },
+                    ],
+                },
+            },
+        },
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema)
+    ctx = {"tenant": {"tenant_id": "6", "username": "lessa"}}
+    toggle = feature_flags.evaluate(name="my_feature", context=ctx, default=False)
+    assert toggle == expected_value
+
+
+def test_get_feature_toggle_propagates_access_denied_error(mocker):
+    # GIVEN a schema fetch that raises a StoreClientError
+    # due to client invalid permissions to fetch from the store
+    err = "An error occurred (AccessDeniedException) when calling the GetConfiguration operation"
+    schema_fetcher = init_fetcher_side_effect(mocker, StoreClientError(err))
+    feature_flags = FeatureFlags(schema_fetcher)
+
+    # WHEN calling evaluate
+    # THEN raise StoreClientError error
+    with pytest.raises(StoreClientError, match="AccessDeniedException") as err:
+        feature_flags.evaluate(name="Foo", default=False)
+
+
+def test_get_configuration_with_envelope_and_raw(mocker):
+    expected_value = True
+    mocked_app_config_schema = {"log_level": "INFO", "features": {"my_feature": {"default": expected_value}}}
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, envelope="features")
+
+    features_config = feature_flags.get_configuration()
+    raw_config = feature_flags.store.get_raw_configuration
+
+    assert "log_level" in raw_config
+    assert "log_level" not in features_config
+
+
+##
+## Inequality test cases
+##
+
+
+# Test not equals
+def test_flags_not_equal_no_match(mocker):
```

Comment:
The same as above, the following 14 test cases test_flag_(not_eq/gt/lt)_(no_match/match) are pretty similar to me. If we could simplify these test cases using `@pytest.mark.parametrize` this will make these test cases much simpler and easier to follow.